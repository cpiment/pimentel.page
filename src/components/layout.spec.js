import React from "react"
import renderer from "react-test-renderer"
import * as Gatsby from 'gatsby';
import Layout from "./layout"

const useStaticQuery = jest.spyOn(Gatsby, 'useStaticQuery');
useStaticQuery.mockImplementation(() => ({
  site: {
    siteMetadata: {
      title: 'pimentel.page',
    },
  },
}));

beforeEach(() => {
    jest.clearAllMocks();
  });

describe("Layout", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Layout pageTitle="Test Page"  isHome="false"><p>Test Page</p></Layout>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})