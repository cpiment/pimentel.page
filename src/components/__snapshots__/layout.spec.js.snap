// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="container"
>
  <title>
    Home
     | 
    pimentel.page
  </title>
  <header
    className="siteTitle"
  >
    <svg
      viewBox="0 0 140 22"
    >
      <text
        x="0"
        y="15"
      >
        pimentel.page
      </text>
    </svg>
  </header>
  <nav>
    <ul
      className="navLinks"
    >
      <li
        className="navLinkItem"
      >
        <a
          className="navLinkText"
          href="/blog"
        >
          Blog
        </a>
      </li>
      <li
        className="navLinkItem"
      >
        <a
          className="navLinkText"
          href="/about"
        >
          About
        </a>
      </li>
      <li
        className="navLinkItem"
      >
        <a
          classname="navLinkText"
          href="https://github.com/cpiment/pimentel.page"
        >
          Github
        </a>
      </li>
    </ul>
  </nav>
  <main>
    <h1
      className="heading"
    >
      Test Page
    </h1>
    <p>
      Test Page
    </p>
  </main>
</div>
`;
